src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,123 @@
+        newUser.setAge(VALID_AGE);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("User with your login has already exists.Try new login!",
+                exception.getMessage());
+    }
+
+    @Test
+    void register_successRegistration_Ok() {
+        User registeredUser = new User();
+        registeredUser.setLogin(VALID_LOGIN);
+        registeredUser.setPassword(VALID_PASSWORD);
+        registeredUser.setAge(VALID_AGE);
+        User actualUser = registrationService.register(registeredUser);
+        assertEquals(registeredUser, actualUser);
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User userNullLogin = new User();
+        userNullLogin.setLogin(null);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userNullLogin);
+        });
+        assertEquals("Login cannot be null or less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userShortLogin = new User();
+        userShortLogin.setLogin(INVALID_BORDERLINE_LOGIN);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userShortLogin);
+        });
+        assertEquals("Login cannot be null or less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userShortPassword = new User();
+        userShortPassword.setLogin(VALID_LOGIN);
+        userShortPassword.setPassword(INVALID_BORDERLINE_PASSWORD);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userShortPassword);
+        });
+        assertEquals("Password cannot be null or less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User userNullPassword = new User();
+        userNullPassword.setLogin(VALID_LOGIN);
+        userNullPassword.setPassword(null);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userNullPassword);
+        });
+        assertEquals("Password cannot be null or less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_lowAge_notOk() {
+        User userLowAge = new User();
+        userLowAge.setLogin(VALID_LOGIN);
+        userLowAge.setPassword(VALID_PASSWORD);
+        userLowAge.setAge(INVALID_BORDERLINE_AGE);
+        userLowAge.setAge(INVALID_NEGATIVE_AGE);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userLowAge);
+        });
+        assertEquals("Age cannot be null or less than 18!", exception.getMessage());
```

Comment:
The exception message should be specific to the failure. Since the age is set to a negative value, the message should reflect that the age cannot be negative.